open-craft/opencraft
# [WIP] Add ability to watch PRs in many repositories

This removes `WATCH_FORK` and `WATCH_ORGANIZATION` and substitutes them by a `WatchedFork` model.

# Testing instructions
1. Don't delete the variables from your .env yet, as they will be used in the migraiton
2. Run the migration. A WatchedFork will be created with the same values you had in your .env
3. Delete WATCH_FORK/WATCH_ORGANIZATION from your .env.
4. In github, use a limited user (not your main one), add it to an „Owners“ team
5. In github, create an `edx/edx-platform` fork and a PR with a small change, and then another fork and another PR
6. Add a WatchedFork for each fork
7. If you want, change some value in the .env (like OPENSTACK_USER) to disable the real deployment of instances, because we just need to test whether it detects PRs and applies settings to instances, not the whole deploy.
8. Invoke watch_pr (in my case it didn't run automatically so I had to invoke manually from shell), and check that it spawned 2 instances, one for each.
9. Repeat the previous steps with some variables embedded in the PR. Or run tests (see next step)
10. Run the tests and check the results. The test `test_tasks.test_watch_several_forks` already does many combinations of variables, in particular checking that a WatchedFork's configuration_source_repo_url is overriden by one defined in the PR, and that the rest of the variables are merged.

# Other notes
- after this, merge https://github.com/open-craft/documentation/pull/224
